integrations/tensorflow/bindings/python/pyiree/tf/support/tf_utils.py

```diff
@@ -248,7 +248,13 @@ def __init__(self, context: rt.SystemContext, f: rt.system_api.BoundFunction):
     self._f = f
 
   def __call__(self, *args):
-    return self._f(*args)
+    new_args = []
+    # For boolean inputs we need to convert to int8.
+    for arg in args:
+      if arg.dtype == np.bool:
+        arg = arg.astype(dtype=np.int8)
+      new_args.append(arg)
```

Comment:
I'm not sure this conversion should happen at this layer. This isn't the only way to call a `rt.system_api.BoundFunction`, so I think this belongs at least as low as that level. I'm not super familiar with the bindings, so it might be more at home in the C++ (e.g. `functional_abi.cc`'s `PyRawPack`).

Also these arguments aren't guaranteed to be numpy arrays, so this should fail the mandlebrot test (which passes `float`s).